src/platform/i386/chal/chal_proto.h

```diff
@@ -41,21 +45,43 @@ struct cap_pgtbl {
 	struct cap_header h;
 	u32_t             refcnt_flags; /* includes refcnt and flags */
 	pgtbl_t           pgtbl;
-	u32_t             lvl;       /* what level are the pgtbl nodes at? */
-	struct cap_pgtbl *parent;    /* if !null, points to parent cap */
-	u64_t             frozen_ts; /* timestamp when frozen is set. */
+	u32_t             lvl;          /* what level are the pgtbl nodes at? */
+	asid_t            asid;         /* hardware context identifier */
```

Comment:
This should only be !0 if the lvl == 0. I might forget this fact later when the node is constructed.

Reply:
This fact does point conceptually for the *right place* for this being in the component creation APIs, eventually.

Reply:
I agree; The issue is just how to add it to the API with the limited number of input arguments we have right now.